setup.py

```diff
@@ -252,7 +253,7 @@ def write_version(filename=os.path.join(*['airflow',
     'qds-sdk>=1.9.6',
     'rednose',
     'requests_mock',
-    'flake8>=3.6.0',
+    'virtualenv',
```

Comment:
Is this module used?

Reply:
Yep. It is used in TestPythonVirtualenvOperator - I had to add it as dependency to make tests work.

However, maybe it's not the best "extra" to put it in. I could for example create a virtualenv extra and add it to "devel_all". What do you think @ashb  ?

Reply:
Probably `devel_ci` then?

Reply:
👍

Reply:
But on the other hand - what if someone actually wants to use the PythonVirtualenvOperator ? Should not they actually have to be given a chance to do pip install -e.[something,something,virtualenv] ? Maybe in this case virtualenv deserves its own extra?

Reply:
Oh python3  https://docs.python.org/3/library/venv.html is built  in

Reply:
Yeah. But venv3 is only available on python3 and not widely used (and has no "virtualenv" script  equivalent). The PythonVirtualenvOperator uses "virtualenv" command to start virtualenv now - so we need to install it also for python3. Maybe when we drop Python 2 (finally!) we can switch the operator to venv and get rid of virtualenv :).

Reply:
This would be great as the virtualenv also introduces another layer of isolation, which isn't necessary and therefore it makes it more complicated. Lets see how the vote will end on Python2 this Sunday :-)

Reply:
Just to clarify - We are not using virtualenv to run the tests themselves :) . In the new image, airflow deps are installed directly in main "python" installation - it's really just needed if you want tor un PythonVirtualenvOperator